sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/bucket.scala

```diff
@@ -56,3 +57,9 @@ private[sql] abstract class BucketedOutputWriterFactory extends OutputWriterFact
       context: TaskAttemptContext): OutputWriter =
     throw new UnsupportedOperationException("use bucket version")
 }
+
+private[sql] object BucketingUtils {
```

Comment:
it might be better to have this in HashPartition rather than BucketingUtils. HashPartitionionig seems like the more general concept.